### PR TITLE
streamingccl: reduce server count in multinode tests

### PR DESCRIPTION
**Pull request by samiskin**
While these tests would pass under stress locally they would fail CI
stress, which may be because we were starting more server processes than
ever before with 4 source nodes, 4 source tenant pods, and 4 destination
nodes.

This PR reduces the node count to 3 (any lower and scatter doesn't
correctly distribute ranges) and only starts a single tenant pod for the
source cluster.

Release justification: test-only change
Release note: None

### CODE REVIEW

**Review by gh-casper — APPROVED** [review]

**Comment by samiskin** [comment]
@stevendanna missed that this got approved last week, and now looks like I need another approval since Casper's account no longer has write access lol

**Review by adityamaru — APPROVED** [review]

**Comment by adityamaru** [comment]
One of the failures in https://github.com/cockroachdb/cockroach/issues/86287 was also under deadlock. It might be worth skipping these heavy-weight tests under deadlock but I'm +1 on checking this in and seeing if reducing the node count brings down the execution time enough.

**Pull request merged**
